pkg/instancetype/controller/vm/controller_test.go

```diff
@@ -482,38 +488,36 @@ var _ = Describe("Instance type and Preference VirtualMachine Controller", func(
 			Expect(vmi.Annotations).ToNot(HaveKey(virtv1.ClusterPreferenceAnnotation))
 		})
 
-		It("should fail to sync if an invalid InstancetypeMatcher Kind is provided", func() {
-			vm.Spec.Instancetype = &virtv1.InstancetypeMatcher{
-				Name: instancetypeObj.Name,
-				Kind: "foobar",
-			}
-
-			_, err := instancetypeController.Sync(vm, vmi)
-			Expect(err).To(MatchError(ContainSubstring("got unexpected kind in InstancetypeMatcher")))
-			testutils.ExpectEvents(recorder, common.FailedCreateVirtualMachineReason)
-		})
-
-		It("should fail to sync if a VirtualMachineInstancetype cannot be found", func() {
-			vm.Spec.Instancetype = &virtv1.InstancetypeMatcher{
-				Name: "foobar",
-				Kind: instancetypeapi.SingularResourceName,
-			}
-
-			_, err := instancetypeController.Sync(vm, vmi)
-			Expect(err).To(MatchError(ContainSubstring("not found")))
-			testutils.ExpectEvents(recorder, common.FailedCreateVirtualMachineReason)
-		})
-
-		It("should fail to sync if a VirtualMachineClusterInstancetype cannot be found", func() {
-			vm.Spec.Instancetype = &virtv1.InstancetypeMatcher{
-				Name: "foobar",
-				Kind: instancetypeapi.ClusterSingularResourceName,
-			}
+		DescribeTable("should fail to sync with FailedFindInstancetype reason",
+			func(matcher *virtv1.InstancetypeMatcher) {
+				vm.Spec.Instancetype = matcher
+				syncVM, err := instancetypeController.Sync(vm, vmi)
+				Expect(syncVM).To(Equal(vm))
+				Expect(err).To(HaveOccurred())
 
-			_, err := instancetypeController.Sync(vm, vmi)
-			Expect(err).To(MatchError(ContainSubstring("not found")))
-			testutils.ExpectEvents(recorder, common.FailedCreateVirtualMachineReason)
-		})
+				var syncErr common.SyncError
+				Expect(errors.As(err, &syncErr)).To(BeTrue())
+				Expect(syncErr.Reason()).To(Equal("FailedFindInstancetype"))
+			},
+			Entry("if an invalid InstancetypeMatcher Kind is provided",
+				&virtv1.InstancetypeMatcher{
+					Name: instancetypeName,
+					Kind: "foobar",
```

Comment:
Don't we want to keep rejecting on this kind of error? It is never going to eventually resolve.